File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs)==0:
            return '' ''
        base = strs[0] 
        for i in range(len(base)):
            for words in strs[1:] :
                if i==len(words) or base[i]!=words[i]:
                    return base[0:i]
        return base          
        
        

File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.empty()){
        return "";
        }
        string base=strs[0];
        for(int i=0;i<base.length();i++)
        {
            for(int words=1;words<strs.size();words++)
            {
                if(i==strs[words].length() || base[i]!=strs[words][i]){
                    return base.substr(0,i);
                }
            }
        }
        return base;
    }
};